### PR TITLE
Refactors and new tests

### PR DESCRIPTION
**Pull request by big-guy**
A little bit of clean-up while going through these classes

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/RepositoryInteractionDependencyResolveIntegrationTest.groovy [hunk 1]
```diff
@@ -376,19 +415,9 @@ class RepositoryInteractionDependencyResolveIntegrationTest extends AbstractHttp
...
                 }
             }
         }
-
-        where:
-        target         | supported
-        'maven'        | false
-        'ivy'          | true
-        'maven-gradle' | false
-        'ivy-gradle'   | false
```
Comment (DPUkyle) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/RepositoryInteractionDependencyResolveIntegrationTest.groovy:392]: Thanks for splitting this test into supported vs. not supported.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewArtifactSelectionIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                }
+            }
+            artifacts {
+                conf file("output")
+                confWithAttributes file("alternative")
+            }
+        """
+        buildFile << """
+            dependencies {
+                resolveConf project(path: ":producer", configuration: "conf")
```
Comment (DPUkyle) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewArtifactSelectionIntegrationTest.groovy:61]: Just to confirm my understanding: by explicitly specifying the `configuration`, we are bypassing variant-aware selection and ignoring attributes. Correct?

**Review by DPUkyle — APPROVED** [review]
LGTM. Please confirm my one comment on my interpretation of `ArtifactViewArtifactSelectionIntegrationTest`.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewArtifactSelectionIntegrationTest.groovy (hunk 2 above)
Comment (big-guy) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewArtifactSelectionIntegrationTest.groovy:61]: Yes, this was a question I had when looking at how selection worked and I didn't see any other test demonstrating this directly.

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**